cloudify_nsx/library/nsx_common.py

```diff
@@ -55,13 +55,12 @@ def _cleanup_properties_list(properties_list):
 
 def _cleanup_if_empty(value):
     """return None if all fileds equal to None, else origin dict"""
-    have_not_none = False
     for key in value:
```

Comment:
Might be better as:
if any(value[key] is not None for key in value):
    return value
else:
    return None

In either case, note that your current method is not actually doing what the docstring says as it will return None for:
value = {
    't1': 0,
    't2': "",
    't3': False,
    't4': datetime.time(0, 0),  # Midnight
}
...even though none of those are actually equal to None